### PR TITLE
Made SMTP, IMAP, and POP3 RFC Compliant

### LINKED ISSUE #238: Send QUIT during SMTP scans.
Also, determine if this is needed for IMAP and POP scans.

### PR DESCRIPTION
**Pull request by ajholland**
Resolves #238 

SMTP server requires Quit command as stated in RFC 5321 Section 3.1

IMAP and POP3 are both ambiguous on whether a server requires a close/quit command because a timeout could be implemented by the server. Either way, sending a quit command probably doesn't hurt.

### CODE REVIEW

File: zlib/grabber.go [hunk 1]
```diff
@@ -514,6 +514,21 @@ func makeGrabber(config *Config) func(*Conn) error {
 			}
 		}
 
+		if config.Banners {
```
Comment (zakird) [inline zlib/grabber.go]: Why is this dependent on banners?

Comment (ajholland) [inline zlib/grabber.go]: Fixed in latest commit, I didn't fully understand banners.

File: zlib/grabber.go [hunk 2]
```diff
@@ -514,6 +514,20 @@ func makeGrabber(config *Config) func(*Conn) error {
 			}
 		}
 
+		if config.SMTP {
+			if err := c.SMTPQuit(); err != nil {
```
Comment (dadrian) [inline zlib/grabber.go]: Can we set `erroredComponent` here in all of the cases, probably to `"quit"`?

Comment (ajholland) [inline zlib/grabber.go]: Yup, pushed

**Pull request merged**